Gradle sync fails when older Kotlin version used

## Expected Behavior
I can successfully apply detekt plugin to an existing project that is using a recent Kotlin version.

## Observed Behavior
Applying detekt 1.19.0-RC1 to a project using Kotlin 1.5.21 fails.

## Steps to Reproduce

1. Clone https://github.com/3flex/detekt-mismatch-kotlin
2. Run `gradlew tasks`
3. See failure:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not create task ':compileKotlin'.
> Could not create task of type 'KotlinCompile'.
   > Could not create an instance of type org.jetbrains.kotlin.gradle.tasks.KotlinJavaToolchainProvider.
      > Could not generate a decorated class for type KotlinJavaToolchainProvider.
         > Cannot have abstract method KotlinJavaToolchain.getJdk().
```

## Context
With this dependency view from the Gradle scan it can be seen that the Kotlin Gradle plugin version dependencies of kotlin-gradle-plugin-api and others are changed from 1.5.21 to 1.5.31 to match the detekt Kotlin dependency version used on detekt Gradle plugin's runtime classpath: https://scans.gradle.com/s/3ohyjhgsh3xig/build-dependencies?dependencies=1.5.21&expandAll&focusedDependencyView=versions&resolutionFilter=WzFd

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used: 1.19.0-RC1
* Version of Gradle used (if applicable): 7.3
* Gradle scan link (add `--scan` option when running the gradle task): https://gradle.com/s/3ohyjhgsh3xig
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): https://github.com/3flex/detekt-mismatch-kotlin

--- a/detekt-gradle-plugin/build.gradle.kts
+++ b/detekt-gradle-plugin/build.gradle.kts
@@ -44,7 +44,7 @@ val pluginCompileOnly: Configuration by configurations.creating
 configurations.compileOnly { extendsFrom(pluginCompileOnly) }
 
 dependencies {
-    implementation(libs.kotlin.gradlePluginApi)
+    compileOnly(libs.kotlin.gradlePluginApi)
     implementation(libs.sarif4k)
 
     pluginCompileOnly(libs.android.gradle)

